### PR TITLE
Domains: Remove old unused DNS components

### PR DESCRIPTION
**Pull request by leonardost**
### Changes proposed in this Pull Request

In #59160 we deployed the redesigned DNS pages to all users. With the redesign, many older components became obsolete, so this PR aims to remove that unused code.

### Testing instructions

- There should be no behavior changes in this PR, but please do a smoke test on the domain DNS settings page to ensure everything is working correctly

### CODE REVIEW

File: client/my-sites/domains/domain-management/name-servers/dns-templates.jsx [hunk 1]
```diff
@@ -1,116 +0,0 @@
-import { Card } from '@automattic/components';
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/name-servers/dns-templates.jsx:1]: This component became `EmailSetup`

File: client/my-sites/domains/domain-management/name-servers/dns-template-selector.jsx [hunk 2]
```diff
@@ -1,28 +0,0 @@
-import { PureComponent } from 'react';
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/name-servers/dns-template-selector.jsx:1]: This was only used in `DnsTemplates`

File: client/my-sites/domains/domain-management/dns/dns-record.jsx [hunk 3]
```diff
@@ -1,124 +0,0 @@
-import { Button, Gridicon } from '@automattic/components';
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/dns/dns-record.jsx:1]: This became `DnsRecordData`

File: client/my-sites/domains/domain-management/dns/dns-list.jsx [hunk 4]
```diff
@@ -1,145 +0,0 @@
-import { localize } from 'i18n-calypso';
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/dns/dns-list.jsx:1]: This became `DnsRecordsList`

File: client/my-sites/domains/domain-management/dns/index.jsx [hunk 5]
```diff
@@ -1,210 +0,0 @@
-import { CompactCard as Card } from '@automattic/components';
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/dns/index.jsx:1]: This component became `DnsRecords`

File: client/my-sites/domains/domain-management/dns/domain-connect-record.jsx [hunk 6]
```diff
@@ -1,141 +0,0 @@
-import { ToggleControl } from '@wordpress/components';
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/dns/domain-connect-record.jsx:1]: This was only used in the `Dns` component

File: client/my-sites/domains/domain-management/dns/dns-records.jsx [hunk 7]
```diff
@@ -135,19 +133,6 @@ class DnsRecords extends Component {
 			</Fragment>
 		);
 	}
-
-	goBack = () => {
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/dns/dns-records.jsx:139]: Unused method

**Comment by gius80** [comment]
LGTM, tested and worked as expected! 🚀

(thanks for the code comments, very useful for the review!)

**Pull request merged**